packages/core/src/components/Form.js

```diff
@@ -266,7 +266,10 @@ export default class Form extends Component {
     }
 
     if (!this.props.noValidate) {
-      let { errors, errorSchema } = this.validate(newFormData);
+      let { errors, errorSchema } = this.validate(
+        newFormData,
+        this.state.schema
```

Comment:
when i refactored the test "should validate formData when the schema is updated" in Form_test.js I noticed that it never called the onError event as I expected, but rather only updated the internal state. I tracked it down to be due to this.validate were still using the previous schema as passed by props. solved this by passing state.schema explicitly.

Reply:
It seems that when I remove this change, the updated "should validate formData when the schema is updated" test is still passing, though. Do you know why this might be the case?

Reply:
really weird, i cannot reproduce the error anymore. must have been something else that messed up the test while i was doing the refactoring. i reverted the change, thanks for checking!